File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int left, best, curr;
        left = best = curr = 0;
        for (int right = 0; right < prices.size(); right++){
            while (prices[left] > prices[right]){
                left++;
            }
        curr = prices[right] - prices[left];
        best = max(best, curr);
        }
    return best;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = best = curr = 0

        for right in range(len(prices)):

            while left != right and  prices[right] < prices[left]:
                left += 1
            curr = prices[right] - prices[left]
            best = max(best, curr)

        return best
